core/vm/gas_table.go

```diff
@@ -261,27 +261,12 @@ func makeGasLog(n uint64) gasFunc {
 		//   and those bytes are charged at LogDataGas (gas per byte) as history growth.
 		// - The remainder of the per-topic cost is attributed to computation (e.g. hashing/bloom work).
 
-		// 32 bytes per topic represents the hash size that gets stored in history.
-		const topicBytes = 32
-
-		var topicHistPer uint64
-		// History growth per topic = topicBytes * LogDataGas
-		if topicHistPer, overflow = math.SafeMul(topicBytes, params.LogDataGas); overflow {
-			return multigas.ZeroGas(), ErrGasUintOverflow
-		}
-		// Sanity check: the per-topic total must be at least the history-growth portion.
-		if params.LogTopicGas < topicHistPer {
-			return multigas.ZeroGas(), fmt.Errorf(
-				"invalid gas params: LogTopicGas(%d) < topicBytes*LogDataGas(%d)",
-				params.LogTopicGas, topicHistPer,
-			)
-		}
 		// Computation per topic = LogTopicGas - (topicBytes * LogDataGas)
-		topicCompPer := params.LogTopicGas - topicHistPer
+		topicCompPer := params.LogTopicGas - params.LogTopicHistoryGas
```

Comment:
This is also a constant, so it should be alongside the other ones. I propose calling it `LogTopicComputationGas`.